tests/unit/datarobot_drum/drum/root_predictors/test_drum_server_utils.py

```diff
@@ -141,3 +152,161 @@ def test_calls_thread_correctly(self, mock_get_command, runner):
         with runner:
             pass
         assert runner.server_thread.command == mock_get_command.return_value
+
+
+class TestStreamingOutput:
+    def test_queue_output_handles_stdout_and_stderr(self):
+        # Create mock objects for stdout, stderr and queue
+        mock_stdout = Mock()
+        mock_stderr = Mock()
+        test_queue = Queue()
+
+        # Configure stdout mock to return a sequence of lines and then stop
+        stdout_lines = [b"stdout line 1\n", b"stdout line 2\n", b""]
+        mock_stdout.readline.side_effect = stdout_lines
+
+        # Configure stderr mock to return a sequence of lines and then stop
+        stderr_lines = [b"stderr line 1\n", b"stderr line 2\n", b""]
+        mock_stderr.readline.side_effect = stderr_lines
+
+        # Call the function
+        _queue_output(mock_stdout, mock_stderr, test_queue)
+
+        # Verify the queue contains all the expected output lines
+        expected_lines = [
```

Comment:
nit.  the ordering in this test feels suspicious.  that's _very_ much an implementation detail that all of stdout is streamed first and then all of stderr.  perhaps this would be more accurate as a `set` ?